spec/lib/repository_cache_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rspec'
+require_relative '../../lib/repository_cache'
+
+describe RepositoryCache do
+  let(:backend) { double("backend").as_null_object }
+  let(:cache) { RepositoryCache.new("example", backend) }
+
+  describe "#cache_key" do
+    it "includes the namespace" do
+      expect(cache.cache_key(:foo)).to eq "foo:example"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.